### DISCUSSION TITLE
Component Library Nextjs

### DISCUSSION THREAD

**Discussion by rtkaaho** [tailwindlabs/tailwindcss#3747] [Help] [answered]
Hi All,

I am currently running into an issue where next production builds are purging css that should be included via the classnames from a component library.

Suppose I have the following example component in an external node_module:

```jsx
import * as React from "react";

interface ButtonProps extends React.HTMLProps<HTMLButtonElement> {
  className?: string;
  type?: "button" | "submit" | "reset"; // standard button jsx type is `string`
}

const ButtonCls = React.forwardRef<HTMLButtonElement, ButtonProps>(
  (props, ref) => {
    return (
      <button
        className="py-2 px-4 bg-red-500 rounded-md"
        {...props}
        ref={ref}
      >
        {props.children}
      </button>
    );
  }
);

ButtonCls.displayName = "Button";

export default ButtonCls;
```

And when bundled via rollup turns into this:

```js
import { forwardRef, createElement } from 'react';

function _extends() {
  _extends = Object.assign || function (target) {
    for (var i = 1; i < arguments.length; i++) {
      var source = arguments[i];

      for (var key in source) {
        if (Object.prototype.hasOwnProperty.call(source, key)) {
          target[key] = source[key];
        }
      }
    }

    return target;
  };

  return _extends.apply(this, arguments);
}

const ButtonCls = /*#__PURE__*/forwardRef((props, ref) => {
  return /*#__PURE__*/createElement("button", _extends({
    className: "py-2 px-4 bg-red-500 rounded-md"
  }, props, {
    ref: ref
  }), props.children);
});
ButtonCls.displayName = "Button";

export { Button, ButtonCls };
//# sourceMappingURL=index.es.js.map

```

Now we can see that the tailwind classes are included in the bundled component.

The problem is that after I import and use the `Button` component in a next.js page and proceed to build the nextjs application any classnames from the `Button` component are not considered in the `purge` and their styles are subsequently removed from the css. 

Is there something I can do to prevent this from happening? Is there another suggested way for creating a component library that uses tailwind? I have been able to make it work with `styled-components` and `twin.macro` but I'd rather not have to include extra tooling.

Also, I have tried adding something like this: `'./node_modules/<path_to_module>/index.es.js'` to the `purge`  option to tell it to look in the node_modules for any styles. But that also didnt seem to work.

Thanks in advance for your help and sorry for the long message!

**Comment by adamwathan** [comment, marked as answer]
Hey! Make sure you are including the paths to your external components in your purge config, that's all you should need to do.

```
module.exports = {
  purge: [
    "./pages/**/*.js",
    "./components/**/*.js",
    "./node_modules/some-library/**/*.js",
  ],
  // ...
}
```

**Reply by rtkaaho** [reply]
Nice! I will give that a try. Thanks @adamwathan . Cheers from KW :)

**Reply by rtkaaho** [reply]
So, as noted in the original post I did try this and it didnt work. However, I did some more digging and forgot that since I am using lerna and yarn workspaces, the component library symlink was getting hoisted up to the root project node_modules.

So anyone in advance that is looking to do something similar to me, just double check to make sure you are referencing the library in the right node_modules folder in the purge config :)

**Reply by IzioDev** [reply]
Oh my god! Thank you so much for this!
This is honestly worth mentioning in the documentation.

**Reply by u12206050** [reply]
This would include the classes of unused components as well though right?
